File: Frontend/app_resources/lib/src/models/Binnacle_model.dart


import 'dart:convert';

import 'package:intl/intl.dart';


BinnacleViewModel binnacleViewFromJson( String str ) => BinnacleViewModel.fromJson(json.decode(str));
class BinnacleViewModel {
  final int? binnacleid;
	final String? monthname;
  final int? year;
	final double? totalhours;
	final bool? status;       
	List<dynamic>? projects;   
  List<DayModel>? days;
  
  BinnacleViewModel({
    this.binnacleid,
    this.monthname,
    this.year,
    this.totalhours,
    this.status,
    this.projects,
    this.days,
  });

  factory BinnacleViewModel.fromJson( Map<String, dynamic> json ) => new BinnacleViewModel(
    binnacleid: json['binnacleid'],
    monthname: json['monthname'],
    year: json['year'],
    totalhours: json['totalhours'],
    status: json['status'],
    projects: (json['projectnames'] as List),
    days: (json['days'] as List).map((e) => dayDataFromJson(e)).toList(),
  );

  Map<String, dynamic> toJson() => {
    "binnacleid": binnacleid,
    "monthname": monthname,
    "year": year,
    "totalhours": totalhours,
    "status": status,
    "projects": projects,
    "days": days,
  };
}

DayModel dayDataFromJson( Map<String, dynamic> json ) => DayModel.fromJson(json);
class DayModel {
  final int? dayid;
	final String? dayname;
	final String? daydate;
	final String? totalhoursday;
	final String? checktime;
	final String? departuretime;
	final String? overtime;
	final String? mealtime;
	final String? summary;
	final String? invoice;

  DayModel({
    this.dayid,
    this.dayname,
    this.daydate,
    this.totalhoursday,
    this.checktime,
    this.departuretime,
    this.overtime,
    this.mealtime,
    this.summary,
    this.invoice,
  });

  factory DayModel.fromJson( Map<String, dynamic> json ) => new DayModel(
    dayid: json['dayid'],
    dayname: json['dayname'],
    daydate: DateFormat("dd-MM-yy").format(DateTime.parse(json['daydate'])),
    totalhoursday: DateFormat("HH:mm::ss").format(DateTime.parse(json['totalhoursday'])),
    checktime: DateFormat("HH:mm:ss").format(DateTime.parse(json['checktime'])),
    departuretime: DateFormat("HH:mm:ss").format(DateTime.parse(json['departuretime'])),
    overtime: DateFormat("HH:mm:ss").format(DateTime.parse(json['overtime'])),
    mealtime: DateFormat("HH:mm:ss").format(DateTime.parse(json['mealtime'])),
    summary: json['summary'],
    invoice: json['invoice'],
  );
}

File: Backend/domain/model/binnacle_model.go
package model

import (
	"time"
)

type Binnacle struct {
	Binnacleid  int `json:id`
	Monthname   string
	Month       int
	Year        int
	Totalhours  time.Time
	Createat    time.Time
	Status      bool
	Projectname string
	Userid      int
}

type CreateBinnacle struct {
	Month  int
	Year   int
	Userid int
	Status bool `json:"-"`
}
type FisnisBinnacle struct {
	Binnacleid int
	Status     bool
	Hourstotal float64
}

type BinnacleView struct {
	BinnacleData BinnacleViewData   `json:"binnacledata"`
	Projectnames []string           `json:"projectnames"`
	Days         []*DayBinnacleView `json:"days"`
}
type BinnacleViewUser struct {
	Binnacleid   int                    `json:"binnacleid"`
	Monthname    string                 `json:"monthname"`
	Year         int                    `json:"year"`
	Totalhours   float64                `json:"totalhours"`
	Status       bool                   `json:"status"`
	Projectnames []string               `json:"projectnames"`
	Days         []*DayBinnacleViewUser `json:"days"`
}

type DayBinnacleView struct {
	Daydata       *Day             `json:"daydata"`
	Checktime     ChecktimeDay     `json:"checktime"`
	Mealtime      Mealtime         `json:"mealtime"`
	Departuretime DeparturetimeDay `json:"departuretime"`
	Overtime      Overtime         `json:"overtime"`
	Activity      ActivityDay      `json:"activity"`
}
type DayBinnacleViewUser struct {
	Dayid         int       `json:"dayid"`
	Dayname       string    `json:"dayname"`
	Daydate       time.Time `json:"daydate"`
	Totalhoursday time.Time `json:"totalhoursday"`
	Checktime     time.Time `json:"checktime"`
	Departuretime time.Time `json:"departuretime"`
	Overtime      time.Time `json:"overtime"`
	Mealtime      time.Time `json:"mealtime"`
	Summary       string    `json:"summary"`
	Invoice       string    `json:"invoice"`
}

type BinnacleViewData struct {
	Binnacleid int
	Monthname  string
	Year       int
	Totalhours float64
	Status     bool
	Userid     int
	Firstname  string
	Lastname   string
	Photouser  string
}

type CreateDay struct {
	Dayid      int
	Binnacleid int
	Month      int
	Year       int
	Projectids []int
}

type Departuretime struct {
	Departuretime time.Time `json:"departuretime"`
	Daydate       time.Time `json:"daydate"`
	Overtime      bool      `json:"overtime"`
	Summary       string    `json:"summary"`
	Invoice       string    `json:"invoice"`
	Userid        int       `json:"userid"`
}
type CreateDeparureTime struct {
	Binnacleid int
	Dayid      int
	Checktime  int
	Mealtime   int
}

type Day struct {
	Dayid         int
	Dayname       string
	Daydate       time.Time
	Createat      time.Time
	Totalhoursday time.Time
	Dailytime     time.Time
}
type ChecktimeDay struct {
	Checktimeid int
	Checktime   time.Time
	Createat    time.Time
	Dayid       int
}
type DeparturetimeDay struct {
	Departuretimeid int
	Departuretime   time.Time
	Createat        time.Time
}
type ActivityDay struct {
	Activityid int
	Summary    string
	invoice    string
}

type Mealtime struct {
	Mealtimeid int
	Mealtime   time.Time `json:"mealtime"`
	Createat   time.Time
	Dayid      int
	Daydate    time.Time `json:"daydate"`
	Userid     int       `json:"userid"`
}
type CreateMealTime struct {
	Binnacleid      int
	Dayid           int
	DeparturetimeId int
}

type Checktime struct {
	Checktimeid int
	Checktime   time.Time
	Createat    time.Time
	Dayid       int
	Userid      int
	Binnacleid  int
	Daydate     time.Time
}

type Overtime struct {
	Overtimeid int
	Overtime   time.Time
	Createat   time.Time
	Dayid      int
}

type Activity struct {
	Activityid int
	Summary    string
	Invoice    string
	Dayid      int
}

type UpdateDay struct {
	Dayid         int
	Daydate       time.Time
	Totalhoursday time.Time
	Dailytime     time.Time
	Checktime     time.Time
	Mealtime      time.Time
	Departuretime time.Time
	Overtime      time.Time
	Summary       string
	Invoice       string
}
